Fix pipeline for pytest with Python 3.7

The build pipeline for pytest with Python 3.7 failed at the stage of installing dependencies, with the following screen output:
```
Collecting xarray>=0.16.1 (from arviz->-r requirements.txt (line 4))
  Downloading xarray-0.20.2-py3-none-any.whl (845 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 845.2/845.2 kB 56.9 MB/s eta 0:00:00
Collecting netcdf4 (from arviz->-r requirements.txt (line 4))
  Downloading netCDF4-1.6.5.tar.gz (764 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 765.0/765.0 kB 55.1 MB/s eta 0:00:00
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'error'
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [37 lines of output]
      Package hdf5 was not found in the pkg-config search path.
      Perhaps you should add the directory containing `hdf5.pc'
      to the PKG_CONFIG_PATH environment variable
      No package 'hdf5' found
      reading from setup.cfg...
      
          HDF5_DIR environment variable not set, checking some standard locations ..
      checking /home/runner/include ...
      hdf5 headers not found in /home/runner/include
      checking /usr/local/include ...
      hdf5 headers not found in /usr/local/include
      checking /sw/include ...
      hdf5 headers not found in /sw/include
      checking /opt/include ...
      hdf5 headers not found in /opt/include
      checking /opt/local/include ...
      hdf5 headers not found in /opt/local/include
      checking /opt/homebrew/include ...
      hdf5 headers not found in /opt/homebrew/include
      checking /usr/include ...
      hdf5 headers not found in /usr/include
      Traceback (most recent call last):
        File "/opt/hostedtoolcache/Python/3.7.17/x[64](https://github.com/CUQI-DTU/CUQIpy/actions/runs/6663819547/job/18110867108#step:4:65)/lib/python3.7/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/opt/hostedtoolcache/Python/3.7.17/x64/lib/python3.7/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/opt/hostedtoolcache/Python/3.7.17/x64/lib/python3.7/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
        File "/tmp/pip-build-env-p24juo15/overlay/lib/python3.7/site-packages/setuptools/build_meta.py", line 341, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
        File "/tmp/pip-build-env-p24juo15/overlay/lib/python3.7/site-packages/setuptools/build_meta.py", line 323, in _get_build_requires
          self.run_setup()
        File "/tmp/pip-build-env-p24juo15/overlay/lib/python3.7/site-packages/setuptools/build_meta.py", line 338, in run_setup
          exec(code, locals())
        File "<string>", line 457, in <module>
        File "<string>", line 3[98](https://github.com/CUQI-DTU/CUQIpy/actions/runs/6663819547/job/18110867108#step:4:99), in _populate_hdf5_info
      ValueError: did not find HDF5 headers
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
Error: Process completed with exit code 1.
```

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -13,7 +13,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest] # Add windows-latest, macos-latest to test on multiple OS
-        python-version: ["3.7", "3.8", "3.9", "3.10"]
+        python-version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
     steps:
       - uses: actions/checkout@v3
       - name: Set up Python ${{ matrix.python-version }}

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -11,7 +11,7 @@ maintainers = [
     {name = "Amal M. Alghamdi", email = "amaal@dtu.dk"}
 ]
 readme = "README.md"
-requires-python = ">=3.7"
+requires-python = ">=3.8"
 license = {file = "LICENSE"}
 dynamic = ["dependencies", "version"]
 

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -8,7 +8,8 @@ description = "Computational Uncertainty Quantification for Inverse problems in 
 maintainers = [
     {name = "Nicolai A. B. Riis", email = "nabr@dtu.dk"},
     {name = "Jakob S. Jørgensen", email = "jakj@dtu.dk"},
-    {name = "Amal M. Alghamdi", email = "amaal@dtu.dk"}
+    {name = "Amal M. Alghamdi", email = "amaal@dtu.dk"},
+    {name = "Chao Zhang", email = "chaz@dtu.dk"}
 ]
 readme = "README.md"
 requires-python = ">=3.8"

